gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-http-get/src/main/java/io/gravitee/plugin/entrypoint/http/get/HttpGetEntrypointConnector.java

```diff
@@ -0,0 +1,310 @@
+                        .map(
+                            message -> {
+                                ctx.putInternalAttribute(ATTR_INTERNAL_LAST_MESSAGE_ID, message.id());
+                                return this.toXmlBuffer(message);
+                            }
+                        )
+                )
+                .concatWith(Flowable.just(Buffer.buffer("</items>")))
+                .concatWith(computePagination(ctx, contentType))
+                .concatWith(Flowable.just(Buffer.buffer("</response>")));
+        } else {
+            return Flowable
+                .just(Buffer.buffer("items="))
+                .concatWith(
+                    ctx
+                        .response()
+                        .messages()
+                        .map(
+                            message -> {
+                                ctx.putInternalAttribute(ATTR_INTERNAL_LAST_MESSAGE_ID, message.id());
+                                return toPlainTextBuffer(message, first.getAndSet(false));
+                            }
+                        )
+                )
+                .concatWith(computePagination(ctx, contentType));
+        }
+    }
+
+    private Buffer toJsonBuffer(Message message, boolean isFirstElement) {
+        JsonObject jsonMessage = new JsonObject();
+
+        if (configuration.isHeadersInPayload()) {
+            JsonObject headers = JsonObject.mapFrom(message.headers());
+            jsonMessage.put("headers", headers);
+        }
+        jsonMessage.put("id", message.id());
+        jsonMessage.put("content", message.content().toString());
+
+        if (configuration.isMetadataInPayload()) {
+            JsonObject metadata = JsonObject.mapFrom(message.metadata());
+            jsonMessage.put("metadata", metadata);
+        }
+
+        String messageString = (!isFirstElement ? "," : "") + jsonMessage.encode();
+        return Buffer.buffer(messageString);
+    }
+
+    private Buffer toXmlBuffer(Message message) {
+        StringBuilder messageBuilder = new StringBuilder();
+        messageBuilder.append("<item>");
+
+        if (configuration.isHeadersInPayload()) {
+            messageBuilder.append("<headers>");
+            message
+                .headers()
+                .toListValuesMap()
+                .forEach(
+                    (header, values) ->
+                        messageBuilder
+                            .append("<")
+                            .append(header)
+                            .append(">")
+                            .append(String.join(",", values))
+                            .append("</")
+                            .append(header)
+                            .append(">")
+                );
+            messageBuilder.append("</headers>");
+        }
+
+        messageBuilder.append("<id>").append(message.id()).append("</id>");
+        messageBuilder.append("<content>").append(message.content().toString()).append("</content>");
```

Comment:
I think it can cause invalid xml if the message content is not properly escaped.